### PR TITLE
Fixed a bug where the  repo:has.description() parameter shows repo name instead of description on Bitbucket server repos

### LINKED ISSUE #46713: repo:has.description parameter shows repo name instead of description on Bitbucket server repos
- **Sourcegraph version:**<v4.2.1
- **Platform information:** Bitbucket Server v7.21.4

#### Steps to reproduce:

1. create repo with a description on [bitbucket.sgdev.org](http://bitbucket.sgdev.org/)

image

2. Use the `repo:has.description(some_description)` search query to get the repo name and description

image

#### Expected behavior:
It is expected that the search query should show the repo description instead.

#### Actual behavior:
It shows the repo name instead of the repo description.

image

If you would like immediate help on this, please email support@sourcegraph.com (you can still create the issue, but there are no [SLAs](https://about.sourcegraph.com/support/) on issues like there are for support requests).

/cc @ryphil

### PR DESCRIPTION
**Pull request by mibali**
fixes #46713 

Previously, we set the description to be the repo name (See [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/repos/bitbucketserver.go?L170)). This made the `repo:has.description` parameter shows repo name instead of the description on Bitbucket server repos.

In this PR, I added a `Description` type for the Repo struct and then modified the `Description` in the `BitbucketServerSource` function to point to the description. This would make the `repo:has.description` parameter to now show the actual description instead of the repo name.

## Test plan
Tested locally.

### CODE REVIEW

**Review by indradhanush — CHANGES_REQUESTED** [review]
Great PR @mibali and thank you so much for the fix! ❤️

The changes look good to me however the tests are failing and would need to be updated. For API tests we use real API response data but save it and reuse it for subsequent test runs. In this case since we're adding a new field that we expect from the API, we should update the tests by adding the `-update` flag to the test command. In this case I can see two relevant tests that need to be updated:

1. `TestClient_LabeledRepos`
1. `TestClient_ProjectRepos`

To update them you will need access to an API token for our test instance which you can find in 1pass. You may run the tests like:

```bash
cd internal/extsvc/bitbucketserver
BITBUCKET_SERVER_TOKEN=<insert-token-here> go test -v -run TestClient_LabeledRepods -update
```

Repeat this for all the other test that need to be updated.

Additionally I would also recommend to add a description to one of the repos in the instance which get fetched from our tests and see if that is reflecting in the newly generated testdata. This should help with validating in our tests that the new changes work.

File: internal/repos/testdata/vcr/path-pattern.yaml [hunk 1]
```diff
@@ -10,69 +10,19 @@ interactions:
...
-      X-Auserid:
-      - "1"
-      X-Ausername:
-      - milton
-      X-Content-Type-Options:
-      - nosniff
-    status: 200 OK
-    code: 200
+      - '@TCJ96Jx619x10993429x1'
+    status: 401 Unauthorized
```
Comment (indradhanush) [inline internal/repos/testdata/vcr/path-pattern.yaml:26]: This doesn't look right to me. These are the files that I updated. Let me take a look.

**Review by indradhanush — CHANGES_REQUESTED** [review]

File: internal/repos/testdata/vcr/path-pattern.yaml (hunk 1 above)
Comment (indradhanush) [reply internal/repos/testdata/vcr/path-pattern.yaml:26]: Okay I fixed the token in the test run and I'm seeing the same timeout error as you. I'll investigate this.

Comment (mibali) [reply internal/repos/testdata/vcr/path-pattern.yaml:26]: Interestingly Buildkite is all green on my end

Comment (indradhanush) [reply internal/repos/testdata/vcr/path-pattern.yaml:26]: Yep it is, but the generated file is not correct. Take a look at the `status: 401 Unauthorized'. This was likely me having a typo in the token. I've fixed it with the correct token and I am now seeing the timeout locally.

File: internal/repos/bitbucketserver_test.go [hunk 2]
```diff
@@ -449,7 +449,7 @@ func TestBitbucketServerSource_ListByProjectKeyAuthentic(t *testing.T) {
 				"SOURCEGRAPH",
 			}
 
-			ctxWithTimeOut, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			ctxWithTimeOut, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (indradhanush) [inline internal/repos/bitbucketserver_test.go:452]:
```suggestion
			ctxWithTimeOut, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

**Review by indradhanush — APPROVED** [review]

**Pull request merged**